packages/components/tabs/src/tab-group.ts

```diff
@@ -43,48 +67,156 @@ export class TabGroup extends ScopedElementsMixin(LitElement) {
+            <span class="indicator" role="presentation"></span>
+            <slot name="tabs" @slotchange=${() => this.#rovingTabindexController.clearElementCache()}></slot>
+            <div
+              id="tabs-popover"
+              ${anchor({ element: this.#moreButton, position: this.vertical ? 'bottom-start' : 'bottom-end' })}
+              @toggle=${this.#onToggle}
+              popover
+              role="listbox"
+              @click=${this.#handleTabChange}
+              part="listbox"
+            >
+              ${this.#allTabs}
+              <span class="indicator-listbox" role="presentation"></span>
+            </div>
+          </div>
+        </div>
+        ${this.#showMore
+          ? html` <sl-button
+              id="more-btn"
+              @click=${this.#onClick}
+              popovertarget="tabs-popover"
+              fill="ghost"
+              variant="primary"
+              size="md"
+              aria-controls="tabs-popover"
+              aria-expanded="false"
+              aria-haspopup="true"
+              aria-label=${msg('Show all')}
+            >
+              <sl-icon name="ellipsis"></sl-icon>
+            </sl-button>`
+          : nothing}
       </div>
       <slot></slot>
     `;
   }
 
   override connectedCallback(): void {
     super.connectedCallback();
+
+    this.#sizeObserver.observe(this);
     this.#updateSlots();
   }
 
+  override disconnectedCallback(): void {
+    this.#sizeObserver.disconnect();
+
+    super.disconnectedCallback();
+  }
+
   override firstUpdated(): void {
     this.#observer = new MutationObserver(this.#handleMutation);
     this.#observer?.observe(this, TabGroup.#observerOptions);
-    setTimeout(() => this.#updateSelectionIndicator(), 100);
+    setTimeout(() => this.#updateSelectionIndicator(), 700);
+  }
+
+  override updated(changes: PropertyValues<this>): void {
+    super.updated(changes);
+
+    if (changes.has('alignment') || changes.has('vertical')) {
+      requestAnimationFrame(() => {
+        this.#updateSelectionIndicator();
+      });
+    }
   }
 
   #updateSlots(): void {
     this.#setupTabs();
     this.#setupPanels();
   }
 
+  #onClick(): void {
+    this.listbox.togglePopover();
+  }
+
+  #onToggle = (event: Event): void => {
+    const indicatorListbox = this.shadowRoot?.querySelector('.indicator-listbox') as HTMLElement;
+
+    this.#rovingTabindexController.clearElementCache();
+
+    requestAnimationFrame(() => {
+      if (!this.listbox || !this.selectedTab) {
+        return;
+      }
+      this.selectedTabInListbox = this.listbox.querySelector(`#${this.selectedTab.id}`) as Tab;
+      indicatorListbox.style.transform = `translateY(${this.selectedTabInListbox?.offsetTop}px)`;
+      indicatorListbox.style.height = `${this.selectedTabInListbox.offsetHeight}px`;
+    });
+
+    if (
+      ((event as ToggleEvent).newState === 'closed' && (event.target as HTMLElement).matches(':popover-open')) ||
```

Comment:
`matches(':popover-open')` only works for non-polyfilled browsers. You need to include the `\:popover-open` selector so it works in older ones as well (please check the polyfill to be sure).